src/util/parser.ts

```diff
@@ -0,0 +1,19 @@
+export const tryParseObject = (data: string | Buffer): Record<string, any> => {
+  try {
+    if (Buffer.isBuffer(data)) {
+      return JSON.parse(data.toString())
+    }
+    const dataString = data.toString()
+    const startIndex = dataString.indexOf('{')
+    const endIndex = dataString.lastIndexOf('}') + 2
+    const messageString = dataString.substr(startIndex, endIndex - startIndex)
```

Comment:
Why is this required? Wondering how performant this is if there are loads of messages?

Reply:
Yep, just hacked this together to get forward but this data receiving is the thing I meant when I mentioned that it would be nice to have a library handle it.
Receiving partial messages or such isn't handled at all (not sure if that can really happen but theoretically it could)